ui/src/commonMain/kotlin/com/materiiapps/gloom/ui/transitions/SlideTransition.kt

```diff
@@ -0,0 +1,47 @@
+package com.materiiapps.gloom.ui.transitions
+
+import androidx.compose.animation.AnimatedContent
+import androidx.compose.animation.AnimatedVisibilityScope
+import androidx.compose.animation.core.FiniteAnimationSpec
+import androidx.compose.animation.core.Spring
+import androidx.compose.animation.core.VisibilityThreshold
+import androidx.compose.animation.core.spring
+import androidx.compose.animation.slideInHorizontally
+import androidx.compose.animation.slideOutHorizontally
+import androidx.compose.animation.togetherWith
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.IntOffset
+import cafe.adriel.voyager.core.screen.Screen
+import cafe.adriel.voyager.core.stack.StackEvent
+import cafe.adriel.voyager.navigator.Navigator
+
+@Composable
+fun SlideTransition(
+    navigator: Navigator,
+    modifier: Modifier = Modifier,
+    animationSpec: FiniteAnimationSpec<IntOffset> = spring(
+        stiffness = Spring.StiffnessMediumLow,
+        visibilityThreshold = IntOffset.VisibilityThreshold
+    ),
+    content: @Composable AnimatedVisibilityScope.(Screen) -> Unit = { it.Content() }
+) {
+    AnimatedContent(
+        targetState = navigator.lastItem,
+        label = "Screen Transition",
+        modifier = modifier,
+        transitionSpec = {
+            val (initialOffset, targetOffset) = when (navigator.lastEvent) {
+                StackEvent.Pop -> ({ size: Int -> -size }) to ({ size: Int -> size })
+                else -> ({ size: Int -> size }) to ({ size: Int -> -size })
+            }
```

Comment:
this seems overcomplicated esp. if you add more branches, somehting like this could work
```kt
val direction = when (navigator.lastEvent) {
  Pop -> -1
  else -> 1
}
```
initialOffset: `{ it * direction }`
targetOffset: `{ it * -direction }`

Reply:
i copied this from voyager internals bc they use an outdated version of compose animation